daemon/start.go

```diff
@@ -152,7 +152,19 @@ func (daemon *Daemon) containerStart(ctx context.Context, daemonCfg *configStore
 		return err
 	}
 
-	spec, err := daemon.createSpec(ctx, daemonCfg, container)
+	mnts, err := daemon.setupContainerDirs(container)
+	if err != nil {
+		return err
+	}
+
+	if m, cleanup, err := daemon.setupMounts(ctx, container); err != nil {
+		return err
+	} else {
```

Comment:
Uneccessary `else`

Reply:
It's actually needed, because `m` and `cleanup` are scoped to the if.

Reply:
[“Programs must be written for people to read, and only incidentally for machines to execute.”
― Harold Abelson, Structure and Interpretation of Computer Programs](https://www.goodreads.com/quotes/9168-programs-must-be-written-for-people-to-read-and-only)

The scoping of `m` is only a problem for the person writing the code. [Indenting error flow](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow) improves the readability of the code. Readability of the code takes priority.

Reply:
Scope can also provide useful information to the reader.
In my view the `else` doesn't decrease readability here and the scope is also a useful information to the reader.
I agree that it's much more readable to indent the error flow and return as early as possible, instead of indenting the whole happy path covering majority of the function. In this case only 2 lines are indented, which are tightly coupled to the result of `setupMounts`.

But, if you're the third reader to agree with this suggestion, then I must accept that it's more readable and my subjective perception of readability is less likely to be shared by an average reader, so I'll just conform 😄.

Changed.